xdsl/dialects/linalg.py

```diff
@@ -936,7 +936,7 @@ def __init__(
         input: SSAValue,
         init: SSAValue,
         permutation: Attribute,
-        result: Attribute | None = None,
+        result: Sequence[Attribute] | None = [],
     ):
         arg_types = NamedOperation.body_arg_types((input, init))
```

Comment:
I would much rather keep the default value None as then it's unambiguously unspecified, whereas the empty list could be provided by the user or the default. One consideration is that if the user did not specify the result then we can actually infer it in most cases. I would recommend inferring it here, for example:
```suggestion
        result: Attribute | None = None,
    ):
    if result is None:
        if isa(input.type, TensorType):
            result = (input.type,)
        else:
            result = ()
    arg_types = NamedOperation.body_arg_types((input, init))
```